src/main/scala/com/tapad/docker/ComposeFile.scala

```diff
@@ -183,7 +186,12 @@ trait ComposeFile extends SettingsHelper with ComposeCustomTagHelpers with Print
    * @param serviceKeys The Docker Compose Yaml representing a service
    * @return PortInfo collection for all defined ports
    */
-  def getPortInfo(serviceKeys: java.util.LinkedHashMap[String, Any]): List[PortInfo] = {
+  def getPortInfo(
+    serviceName: String,
+    serviceKeys: java.util.LinkedHashMap[String, Any],
+    useStatic: Boolean,
+    usedStaticPorts: scala.collection.mutable.Set[String]
```

Comment:
`usedStaticPorts` is a set for keeping track of used static port mappings across services in the Compose file. The caller function `processCustomTags` holds this data structure and passes it down to `getPortInfo` to have the set updated while iterating every service.